be/src/olap/comparison_predicate.h

```diff
@@ -232,8 +251,16 @@ class ComparisonPredicateBase : public ColumnPredicate {
             if constexpr (std::is_same_v<T, StringRef>) {
                 return bf->test_bytes(_value.data, _value.size);
             } else {
-                return bf->test_bytes(const_cast<char*>(reinterpret_cast<const char*>(&_value)),
-                                      sizeof(WarpperFieldType));
+                // DecimalV2 using decimal12_t in bloom filter, should convert value to decimal12_t
+                if constexpr (Type == PrimitiveType::TYPE_DECIMALV2) {
+                    decimal12_t decimal12_t_val(_value.int_value(), _value.frac_value());
+                    return bf->test_bytes(
```

Comment:
what about datev1 and datetimev1， do they also need convert to storage type here? if not please add comment here to explain the reason.